Mathlib/AlgebraicTopology/SimplicialSet/Truncated.lean

```diff
@@ -0,0 +1,222 @@
+/-
+Copyright (c) 2024 Emily Riehl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Emily Riehl
+-/
+import Mathlib.AlgebraicTopology.SimplicialSet.HomotopyCat
+
+/-!
+# 2-truncated Strict Segal simplicial sets
+
+This collects some API which will ultimately be deployed elsewhere
+-/
+
+universe v u
+
+namespace SSet
+
+namespace Truncated
+
+open CategoryTheory Category Functor Simplicial SimplexCategory Opposite
+
+local macro:1000 (priority := high) X:term " _[" n:term "]₂" : term =>
+    `(($X : SSet.Truncated 2).obj (Opposite.op ⟨SimplexCategory.mk $n, by decide⟩))
+
+set_option quotPrecheck false
+local macro:max (priority := high) "[" n:term "]₂" : term =>
+  `((⟨SimplexCategory.mk $n, by decide⟩ : SimplexCategory.Truncated 2))
+
+section
+
+variable (X : SSet.Truncated.{u} 2)
+
+/-- A path in a 2-truncated simplicial set `X` of length `n` is a directed path of `n` edges.-/
+@[ext]
+structure Path₂ (n : ℕ) where
```

Comment:
One approach may be to define `SSet.Truncated.Path₁` which would map `SSet.Truncated.{u} 1` and `n : ℕ` to a type, as we only need to know `0` and `1` simplices in order to define paths. Then, `SSet.Path` and `SSet.Truncated.Path` (for any `X : SSet.Truncated (i + 1)`) could be defined as an abbrev in terms of this `SSet.Truncated.Path₁`. (This is roughly what I had pushed for in the case of `SSet.HomotopyCategory`.)
This type of issues will appear anytime we have constructions on simplicial sets which involve only what happens in lower dimensions: define it first in terms of `SSet.Truncated i` for the most optimal `i`, etc.

Reply:
Here, we could define `SSet.Truncated.StrictSegal X` for any `X : SSet.Truncated (i + 1)`. (There will be some duplication of API with `SSet.StrictSegal` though.)